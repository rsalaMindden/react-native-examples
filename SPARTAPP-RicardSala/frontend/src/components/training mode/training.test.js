import React from "react";
import Training from "./training";
import { render,fireEvent }from '../../utils/tetsUtils/test.utils'


const navigation={
    push:jest.fn()
};
   
    const route={
        params:{
            training:[{name: 'FOR TIME',
                exercises: [{
                  exercise: { name: 'BURPEES',
                    description: 'asdasd',
                    picture: 'asd',
                    done:false,
                    },
                  reps: 10,
                }],
              }]}
        };


describe('Given a Training component',()=>{
    describe('When it is render',()=>{
        let screen;
        beforeEach(()=>{
            screen=render(<Training navigation={navigation} route={route}></Training>);
        });
        test('Then should match tHe snapshot',()=>{
            expect(screen).toMatchSnapshot();
          });
          describe('And sign up button is clicked', () => {
            test('Then navigation.push should have been called with Register', () => {
              const Button = screen.getByTestId('HomePage');
              fireEvent.press(Button);
              expect(navigation.push).toHaveBeenCalledWith('HomePage');
            });
          });
      
        });
    });
 