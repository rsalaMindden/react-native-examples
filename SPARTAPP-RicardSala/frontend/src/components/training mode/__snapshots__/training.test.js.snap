// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given a Training component When it is render Then should match the snapshot 1`] = `undefined`;
